cpp/demo/hyperelasticity/main.cpp

```diff
@@ -1,6 +1,19 @@
-// Hyperelasticity (C++)
-// =========================
-// .. code-block:: cpp
+// ---
+// jupyter:
+//   jupytext:
+//     text_representation:
+//       extension: .cpp
+//       format_name: light
+//       format_version: '1.5'
+//       jupytext_version: 1.16.1
+//   kernelspec:
+//     display_name: C++17
+//     language: C++17
```

Comment:
Can we do C++20 here? That is the minimal required standard we have for the actual code.

Reply:
Specifying kernel_spec is not necessary. This is only needed when this file is opened in a notebook environment, then kernel_spec will be used for a kernel with interactive compiler.
I was trying [xeus-cling](https://github.com/jupyter-xeus/xeus-cling) locally and this jupytext metadata is a remant from that explore. 

One could have just got hold of `main.cpp` file and run

```bash
~/.local/bin/jupytext --set-formats cpp,md main.cpp
```
to bind `.cpp` with a `.md` file, which results in

```
---
jupyter:
  jupytext:
    cell_metadata_filter: -all
    formats: cpp,md
    main_language: c++
    text_representation:
      extension: .md
      format_name: markdown
      format_version: '1.3'
      jupytext_version: 1.16.1
---
```
in `.md` file and 

```
// ---
// jupyter:
//   jupytext:
//     cell_metadata_filter: -all
//     formats: cpp:light,md
//     text_representation:
//       extension: .cpp
//       format_name: light
//       format_version: '1.5'
//       jupytext_version: 1.16.1
// ---
```
`.cpp` file.

https://jupytext.readthedocs.io/en/latest/advanced-options.html

I will simply remove all kernel_spec.

Reply:
Actually only

```bash
~/.local/bin/jupytext --set-formats cpp main.cpp
```
is needed for each `main.cpp` file to populate with jupytext header, which will then be used later to convert to `.md`

Reply:
I was presuming that this jupytext header is needed for reading and then converting to `.md` files.
It turns out that all these specifications are only for connecting `.md` or `.cpp` files to notebook files.

All these are deleted.

I didn't yet try xeus-cling with dolfinx.

Reply:
Interesting, so perhaps we can also reduce or remove them in the .py files.